__tests__/exchanges/kucoin-client.spec.js

```diff
@@ -0,0 +1,61 @@
+const { testClient } = require("../test-runner");
+const KucoinClient = require("../../src/exchanges/kucoin-client");
+
+testClient({
+  clientFactory: () => new KucoinClient(),
+  clientName: "KucoinClient",
+  exchangeName: "KuCoin",
+  markets: [
+    {
+      id: "BTC-USDT",
+      base: "BTC",
+      quote: "USDT",
+    },
+  ],
+
+  testConnectEvents: false,
```

Comment:
We should enable these event tests.  To make them pass, you will need to add a null check of `this._wss` to line 131 of `basic-client.js`.  

Prior to the KuClint client, `_connect` maintained an invariant of synchronously assigning `this._wss`.  

I think maintaining this would be a challenge since we need the results of the async HTTP request to construct the URI for the WS path. The remainder of WS construction relies on that path. While it would be nice to be able to assign _wss synchronously and only defer the connect part until the path is fully constructed it would likely require more substantial changes.